src/Microsoft.AspNetCore.Mvc.Core/Internal/MvcCoreLoggerExtensions.cs

```diff
@@ -184,6 +189,26 @@ static MvcCoreLoggerExtensions()
                 5,
                 "Could not find an output formatter based on content negotiation. Accepted types were ({AcceptTypes})");
 
+            _inputFormatterSelected = LoggerMessage.Define<IInputFormatter, string>(
+                LogLevel.Debug,
+                1,
+                "Selected input formatter '{InputFormatter}' for content type '{ContentType}'.");
+
+            _inputFormatterRejected = LoggerMessage.Define<IInputFormatter, string>(
+                LogLevel.Debug,
+                2,
+                "Rejected input formatter '{InputFormatter}' for content type '{ContentType}'.");
+
+            _noInputFormatterSelected = LoggerMessage.Define<string>(
+                LogLevel.Debug,
+                3,
+                "No input formatter was found to support the content type '{ContentType}' for use with the [FromBody] attribute.");
+
+            _removeFromBodyAttribute = LoggerMessage.Define<string, string>(
+                LogLevel.Debug,
+                4,
+                "To use model binding, remove the [FromBody] attribute from '{ModelName}' with model type '{ModelType}'.");
```

Comment:
cc @pranavkm @Eilon Is this the correct way to convey the information? I am not sure if and how I can access parameter name.